DIPs/DIP1NNN-CrazyPython.md

```diff
@@ -0,0 +1,283 @@
+# Dynamic-size static arrays for D
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | (number/id -- assigned by DIP Manager)                          |
+| Review Count:   | 0 (edited by DIP Manager)                                       |
+| Author:         | James Lu (jamtlu@gmail.com)                                     |
+| Status:         | Draft                                                           |
+## Abstract
+Often, it is necessary to allocate a data structure on the stack based on a
+size known at runtime, and possibly return that data structure. This proposal
+suggests adding dynamic-size static arrays to D. The goal here is an
+easy-to-use dynamically sized array without a GC. It should give the programmer
+the tools to avoid dynamic memory allocation.
+
+## Motivation
+`@nogc` programming is useful in many contexts. For example, currently WASM
+does not have a garbage collector, limiting D applications to `-betterC`
+mode. 
+
+It is often important to be able to quickly write `@nogc` code for
+performance-critical paths. After identifying hot functions using a profile,
+one should be able to alter a dynamic array that is only used within a single
+function to be allocated and deallocated automatically with `@nogc`.
+
+What is missing is an ergonomic way to dynamically allocate a static array on
+the stack built-in to D. You can take slices of static array as parameters
+
+```dlang
+/** A well-known algorithm for finding the topological sorting of a (possibly
+cyclic) directed graph. It deletes edges from its input as it runs, meaning
+adjacency_list is modified from within the function body.
+
+Because dynamic-size static arrays are value types, even when the function body
+modifies adjacency_list, the caller does not see the modifications.
+*/
+@nogc auto kahnsAlgorithm(int[][?] adjacency_list);
```

Comment:
I think this needs to better illustrate the benefit over having `adjacency_list` be a regular static array or a slice of one.